fix tp-link jetstream FDB discovery

Npeca75 opened the pull request:
As expected, TP-LINK have they own way of representing SNMP data
this time, q-bridge-mib ...
port indexes are sent as physical port number, so it need to be find as gigabitEthernet1/0/[X] for proper 'port_id' assignment

fix #15594 

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

Npeca75 commented:
![lnms1](https://user-images.githubusercontent.com/59750439/189046343-ac2d5f50-185c-4d2a-a57d-27b79fedc809.png)

Npeca75 commented:
![lnms1](https://user-images.githubusercontent.com/59750439/189047686-0eb18288-7266-4108-9f07-fe4a3bf2acf3.png)

Jellyfrog commented:
Is fdb table part of tests? Does something happen if you generate new test data?

Npeca75 commented:
> Is fdb table part of tests? Does something happen if you generate new test data?

hi @Jellyfrog 

looked at few jetstream snmprec which contains FDB data, and no, from my oppinion, it is not used / recorded in json
recreated 2 variants with 
```
./scripts/save-test-data.php -o jetstream -v t1600g-28ts
./scripts/save-test-data.php -o jetstream -v ipv6
```

here is output

```
librenms@soada-srv-b:~$ git status
On branch fdb-jetstream
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   tests/data/jetstream_ipv6.json
        modified:   tests/data/jetstream_t1600g-28ts.json

no changes added to commit (use "git add" and/or "git commit -a")
librenms@soada-srv-b:~$ git diff tests/data/jetstream_t1600g-28ts.json
diff --git a/tests/data/jetstream_t1600g-28ts.json b/tests/data/jetstream_t1600g-28ts.json
index cb6b46e00..0e9b83360 100644
--- a/tests/data/jetstream_t1600g-28ts.json
+++ b/tests/data/jetstream_t1600g-28ts.json
@@ -10,11 +10,11 @@
                     "version": "3.0.6 Build 20200805 Rel.55968",
                     "hardware": "T1600G-28TS 3.0",
                     "features": null,
+                    "location": "<private>",
                     "os": "jetstream",
                     "type": "network",
                     "serial": "2199797000735",
-                    "icon": "tplink.svg",
-                    "location": "<private>"
+                    "icon": "tplink.svg"
                 }
             ]
         },
librenms@soada-srv-b:~$ git diff tests/data/jetstream_ipv6.json
diff --git a/tests/data/jetstream_ipv6.json b/tests/data/jetstream_ipv6.json
index 7b3b8a1ef..243f7ff8d 100644
--- a/tests/data/jetstream_ipv6.json
+++ b/tests/data/jetstream_ipv6.json
@@ -10,11 +10,11 @@
                     "version": "3.0.6 Build 20200805 Rel.55968",
                     "hardware": "T1600G-28TS 3.0",
                     "features": null,
+                    "location": "<private>",
                     "os": "jetstream",
                     "type": "network",
                     "serial": "2195441000301",
-                    "icon": "tplink.svg",
-                    "location": "<private>"
+                    "icon": "tplink.svg"
                 }
             ]
         },
librenms@soada-srv-b:~$ 
```

nothing about newly recorded FDB

PipoCanaja commented:
> > Is fdb table part of tests? Does something happen if you generate new test data?
> 
...
> nothing about newly recorded FDB

Correct, FDB is skipped for the moment. So nothing else to do here for tests: 
in : LibreNMS/Util/ModuleTestHelper.php 
```
@@ -55,7 +55,9 @@ class ModuleTestHelper
     // Definitions
     // ignore these when dumping all modules
     private $exclude_from_all = ['arp-table', 'fdb-table'];
```

PipoCanaja commented on includes/discovery/fdb-table/jetstream.inc.php:
```diff
@@ -0,0 +1,46 @@
+        foreach ($oidData as $mac => $macData) {
+            $port = $macData['dot1qTpFdbPort'];
+            //try both variation with & without space
+            $port_id = find_port_id('gigabitEthernet 1/0/' . $port, 'gigabitEthernet1/0/' . $port, $device['device_id']) ?? 0;
```
no chance to have a stack here ? with the "1" hardcoded, won't do good on a stack for sure :)

Npeca75 replied:
if you look at current situation, this solution is 100% better than current one which is 0% recognized port :(

PipoCanaja replied:
that was an open question :) if you tell me jetstream does not support stacking, then this is good to go. And definitly better than current I can imagine :)

murrant replied:
https://static.tp-link.com/res/down/doc/T1700G-28TQ(UN)_V1_UG.pdf

Npeca75 replied:
> https://static.tp-link.com/res/down/doc/T1700G-28TQ(UN)_V1_UG.pdf

Ok mr @murrant 

please look at this:
```

SNMP[.[0;36m'/usr/bin/snmpbulkwalk' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/tplink' '-v2c' '-c' 'COMMUNITY' '-OQXUte' '-Cc' '-Os' 'udp:HOSTNAME:161' 'Q-BRIDGE-MIB::dot1qTpFdbPort'.[0m]

dot1qTpFdbPort[44][74:da:88:4a:ef:ec] = 16
dot1qTpFdbPort[44][a8:5e:45:b5:5f:9c] = 12
dot1qTpFdbPort[100][2:ae:24:96:30:cb] = 24
dot1qTpFdbPort[100][4c:eb:bd:18:c2:4b] = 24
dot1qTpFdbPort[100][9c:7b:ef:5:15:8c] = 24
dot1qTpFdbPort[100][9c:7b:ef:88:f5:10] = 22
dot1qTpFdbPort[200][0:9:0:0:1:3] = 5
dot1qTpFdbPort[200][0:9:0:0:1:5] = 24
```
Ok, Vlan info is in the middle, [xxx], so far so good

but, problem is here:
```
SNMP[.[0;36m'/usr/bin/snmpbulkwalk' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/tplink' '-v2c' '-c' 'COMMUNITY' '-OQXUte' '-Cc' '-Os' 'udp:HOSTNAME:161' 'BRIDGE-MIB::dot1dTpFdbPort'.[0m]
dot1dTpFdbPort[1:0:5f:67:75:df].200 = 49176
dot1dTpFdbPort[1:2:a8:0:5:0].0 = 49176
dot1dTpFdbPort[1:2:a8:0:6:0].1 = 49176
dot1dTpFdbPort[1:2:ae:24:96:30].203 = 49176
dot1dTpFdbPort[1:28:ee:52:f5:92].70 = 49174
dot1dTpFdbPort[1:40:61:86:d:31].0 = 49176
```

this one will give back ifIndex as you see, but only, ONLY for access ports!!! trunk ports are left out

so, could we agreed in next solution ?

1. try to find MAC in bridge-mib (access port)
2. if not, then try to guess (hardcoded 1/0/x)

is this acceptable ?
